pkg/workflows/builder_test.go

```diff
@@ -0,0 +1,300 @@
+package workflows_test
+
+import (
+	_ "embed"
+	"encoding/json"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"sigs.k8s.io/yaml"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities"
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities/consensus/ocr3"
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities/consensus/ocr3/ocr3cap"
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities/targets/chainwriter"
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities/targets/chainwriter/chainwritercap"
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities/triggers/streams"
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities/triggers/streams/streamscap"
+	"github.com/smartcontractkit/chainlink-common/pkg/workflows"
+	"github.com/smartcontractkit/chainlink-common/pkg/workflows/testdata/fixtures/capabilities/notstreams"
+	"github.com/smartcontractkit/chainlink-common/pkg/workflows/testdata/fixtures/capabilities/notstreams/notstreamscap"
+)
+
+// Note that the set of tests in this file cover the conversion from existing YAM -> code
+// along with testing the structure of what is generated from the builders.
+// This implicitly tests the code generators functionally, as the generated code is used in the tests.
+
+type Config struct {
+	Workflow    workflows.NewWorkflowParams
+	Streams     *streams.TriggerConfig
+	Ocr         *ocr3.ConsensusConfig
+	ChainWriter *chainwriter.TargetConfig
+	TargetChain string
+}
+
+func NewWorkflowSpec(rawConfig []byte) (workflows.WorkflowSpec, error) {
+	conf := Config{}
+	if err := yaml.Unmarshal(rawConfig, &conf); err != nil {
+		return workflows.WorkflowSpec{}, err
+	}
+
+	workflow := workflows.NewWorkflow(conf.Workflow)
+	streamsTrigger := streamscap.NewTrigger(workflow, *conf.Streams)
+
+	ocrInput := ocr3cap.ConsensusInput{Observations: workflows.ListOf[streams.Feed](streamsTrigger)}
+	consensus := ocr3cap.NewConsensus(workflow, "ccip_feeds", ocrInput, *conf.Ocr)
+
+	input := chainwritercap.TargetInput{SignedReport: consensus}
+	chainwritercap.NewTarget(workflow, conf.TargetChain, input, *conf.ChainWriter)
```

Comment:
If we are able to co-locate some of the generated types, then we could define these constructor/adders as methods on the config types in order to reduce verbosity:
```suggestion
	workflow := workflows.NewWorkflow(conf.Workflow)
	streamsTrigger := conf.Streams.New(workflow)

	ocrInput := ocr3cap.ConsensusInput{Observations: workflows.ListOf[streams.Feed](streamsTrigger)}
	consensus := conf.Ocr.New(workflow, "ccip_feeds", ocrInput)

	input := chainwritercap.TargetInput{SignedReport: consensus}
	conf.ChainWriter.New(workflow, conf.TargetChain, input)
```